godot-core/src/registry.rs

```diff
@@ -20,8 +20,11 @@ use crate::builtin::StringName;
 use crate::out;
 use std::any::Any;
 use std::collections::HashMap;
+use std::sync::{Mutex, MutexGuard, TryLockError};
 use std::{fmt, ptr};
 
+static LOADED_CLASSES: Mutex<Option<HashMap<InitLevel, Vec<ClassName>>>> = Mutex::new(None);
```

Comment:
I know why it's needed, but I _do_ hate Rust's excessive boilerplate with global variables...

Do you know if this can ever be accessed concurrently? Probably only in case of a Godot bug?

It's probably better to be safe and sorry, since this is not a critical path; as such I appreciate also your panicking and descriptive errors on unlock failure further below.